frontend/web/components/AnnouncementPerPage.tsx

```diff
@@ -31,16 +36,51 @@ const AnnouncementPerPage: FC<AnnouncementPerPageType> = ({ pathname }) => {
     Utils.getFlagsmithHasFeature('announcement_per_page') &&
     announcementPerPageValue?.pages?.length > 0
 
+  const convertObjectValuesToStrings = (
+    obj: AnnouncementPerPageValueType['params'],
+  ) => {
+    const newObj = {}
+    for (const key in obj) {
+      if (obj.hasOwnProperty(key)) {
+        newObj[key] = String(obj[key])
+      }
+    }
+    return newObj
+  }
+
   const announcementInPage = announcementPerPageValue?.pages?.some((page) => {
     if (Object.keys(routes).includes(page)) {
-      return !!matchPath(pathname, {
+      const match = matchPath(pathname, {
         exact: false,
         path: routes[page],
         strict: false,
       })
+
+      if (match) {
+        const annParams =
+          convertObjectValuesToStrings(announcementPerPageValue?.params) || {}
```

Comment:
Is any of the convert objects to strings necessary? Can't you just adjust line 67 to 

```
   return Object.keys(obj1).every((key) => {
            return `${obj1[key]}` === `${obj2[key]}`
          })
```

Reply:
Yes, done